### ISSUE TITLE
Issue with permissions?

### ISSUE THREAD

**Issue by gpatrick2** [gctools-outilsgc/gcconnex#1723]
## User Story for the issue in the format of "As a user of type AAA, I need XXX so that I can YYY."
As a developer, I need to access a list of all blogs which are part of certain groups, so that I can query each blog and check if they mention topics I'm interested in monitoring.

## Details on issue or enhancement

- I'm doing a request to the get.groupblogs API and I noticed that for at least one instance the system is returning me a "group was not found" response for a group I can see exist
- I can see the group exist both through the GCCollab website and through a request to the get.group API using the same GUID
- The group in question is: https://gccollab.ca/groups/profile/85014/the-policy-community-la-communaute-des-politiques
- I've looked at the source code, and the only difference I can see is that the get_group function ignores permissions, while the get_group_blogs function does not. 
- The source code for ELGG also mentions that the get_entity function will only return a result if the entity exist and if I have permission to it. Knowing that the group exist, this leads me to believe the problem is related to permissions.
- I can get information about blogs from other groups using the same API(get.groupblogs) with different GUIDs
- The problem is the group is open and I can view the group and their blogs using the URL above
- Is their some other permission which I'm not seeing which is restricting access to information about a group's blog which I'm required to provide for certain blogs or groups?

## For the development team
- [ ] Issue user story documented
- [ ] UX input received
- [ ] Design completed
- [ ] Design validated by business team / UX
- [ ] Code review completed by peer
- [ ] Issue closing comment references any duplicate or connected issues or pull requests
- [ ] Issue closed

**Comment by gpatrick2** [comment]
I've done a little bit more testing with the the APIs for groups, and found that the same issue happens when requesting any information about events, files, discussions or other items that belong to groups and not blogs only. 

Some groups I'm allowed to receive the information for these items some I am not. All groups I've tested these APIs on are groups which are open, I've found a case where I am a member of one of these groups, but the API won't allow me permission to these items. I've found at least one case where I am not a member of a group and the API allows me permission to these items.
